pkg/migrate/helpers.go

```diff
@@ -0,0 +1,454 @@
+			"regions": []map[string]interface{}{
+				{
+					"namespace":     namespace,
+					"cloudProvider": cloudProvider,
+					"code":          cloudRegion,
+					"nodes":         sts.Spec.Replicas,
+					"domain":        "",
+				},
+			},
+			"dataStore": map[string]interface{}{
+				"volumeClaimTemplate": map[string]interface{}{
+					"metadata": map[string]interface{}{
+						"name": "datadir",
+					},
+				},
+			},
+			"ports": map[string]interface{}{
+				"grpcPort": input.grpcPort,
+				"httpPort": input.httpPort,
+				"sqlPort":  input.sqlPort,
+			},
+			"certificates": map[string]interface{}{
+				"externalCertificates": map[string]interface{}{
+					"caConfigMapName":         sts.Name + "-ca",
+					"nodeSecretName":          sts.Name + "-node-certs",
+					"rootSqlClientSecretName": sts.Name + "-client-certs",
+				},
+			},
+			"affinity":                      sts.Spec.Template.Spec.Affinity,
+			"nodeSelector":                  sts.Spec.Template.Spec.NodeSelector,
+			"tolerations":                   sts.Spec.Template.Spec.Tolerations,
+			"terminationGracePeriodSeconds": sts.Spec.Template.Spec.TerminationGracePeriodSeconds,
+			"loggingConfigMapName":          input.loggingConfigMap,
+			"env":                           sts.Spec.Template.Spec.Containers[0].Env,
+		},
+	}
+}
+
+func generateParsedMigrationInput(
+	ctx context.Context,
+	clientset kubernetes.Interface,
+	sts *appsv1.StatefulSet) (parsedMigrationInput, error) {
+	var startCmd string
+	var envVars = make(map[string]string)
+	var parsedInput = parsedMigrationInput{
+		tlsEnabled: true,
+	}
+
+	// In the public Helm chart, logging configuration is provided as a secret to the StatefulSet.
+	// However, in the Cockroach Enterprise Operator, it is supplied as a ConfigMap.
+	for _, vol := range sts.Spec.Template.Spec.Volumes {
+		if vol.Name == "log-config" {
+			if vol.Secret != nil {
+				parsedInput.loggingConfigMap = vol.Secret.SecretName
+				if err := ConvertSecretToConfigMap(ctx, clientset, sts.Namespace, parsedInput.loggingConfigMap); err != nil {
+					return parsedInput, err
+				}
+			}
+		}
+	}
+
+	for _, c := range sts.Spec.Template.Spec.Containers {
+		if c.Name == "db" {
+			startCmd = c.Args[2]
+			for i := range c.Env {
+				envVars[c.Env[i].Name] = c.Env[i].Value
+			}
+		}
+	}
+
+	if err := extractJoinStringAndFlags(&parsedInput, strings.Fields(startCmd), envVars); err != nil {
+		return parsedInput, err
+	}
+
+	return parsedInput, nil
+}
+
+// extractJoinStringAndFlags parses the command arguments, extracts the --join string, and replaces env variables.
+func extractJoinStringAndFlags(
+	parsedInput *parsedMigrationInput,
+	args []string,
+	envVars map[string]string) error {
+
+	flags := make(map[string]string)
+	// Regular expression to match flags (e.g., --advertise-host=something)
+	flagRegex := regexp.MustCompile(`--([\w-]+)=(.*)`)
+
+	for _, arg := range args {
+		arg = replaceEnvVars(arg, envVars) // Replace env variables
```

Comment:
Why do we need to replace environment variables here? Can we copy environment variables to the operator manifest instead? Also, no need for a "Replace env variables" comment when we're calling a function called `replaceEnvVars`.

Reply:
I tested it before I have added env vars. So yes now we are adding the env vars so yes we can get rid of this func.